File: Codechef/Beginner-Problems/TLG.c
#include <stdio.h>

int main(void) {
	// your code goes here
	int n,x,y;
	int winner;
	int lead=0;
	int num1=0;
	int num2=0;
	scanf("%d",&n);
	for(int i=0;i<n;i++)
	{
	   scanf("%d %d",&x,&y);
	   num1+=x;
	   num2+=y;
	   if(num1>num2)
	   {
	       if((num1-num2)>lead)
	       {
	           lead=num1-num2;
	           winner=1;
	       }
	   }
	   else
	   {
	       if((num2-num1)>lead)
	       {
	           lead=num2-num1;
	           winner=2;
	       }
	   }
	}
	printf("%d %d",winner,lead);
	return 0;
}

File: Codechef/Beginner-Problems/TLG.py
# cook your dish here
n=int(input())
d=list();
num1=0;
num2=0;
f=tuple();
for i in range(0,n):
    x,y=map(int,input().split())
    num1+=x;
    num2+=y;
    if(num1>num2):
        f=(num1-num2,1);
        d.append(f);
    else:
        f=(num2-num1,2);
        d.append(f);
d=sorted(d,reverse=True)
print(d[0][1],d[0][0]);
